### DISCUSSION TITLE
Trouble with wbt_lidar_classification()

### DISCUSSION THREAD

**Discussion by mereiser** [opengeos/whiteboxR#87] [Q&A] [answered]
Hello!

I am trying to use the function `wbt_lidar_clasify_subset()` to classify the ground points in an unclassified point cloud using a .las file of just my classified ground points.  However, regardless of how I assign my ground points I get the error:

```
Error running WhiteboxTools (LidarClassifySubset)
  whitebox.exe_path: "Path/4.2/whitebox/WBT/whitebox_tools.exe"; File exists? TRUE
  Arguments: --run=LidarClassifySubset  --base="/Path/all_points.las" --subset="/Path/las_gr.las" --output="/Path/output.las" --subset_class=2 -v

System command had status 101
**********************************
* Welcome to LidarClassifySubset *
* Powered by WhiteboxTools       *
* www.whiteboxgeo.com            *
**********************************
Reading input files...
Performing analysis...
Creating tree...
Performing analysis...
Progress: 0%
Progress: 1%
thread 'main' panicked at 'index out of bounds: the len is 1458 but the index is 1458', whitebox-lidar\src\las.rs:461:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
lidar_classify_subset - Elapsed Time: NA [did not run]
Warning message:
In length(ret) > 0 && nchar(ret) > 0 :
  'length(x) = 13 > 1' in coercion to 'logical(1)'
```

Every time I've tried this function I get the same error message, that the "len" has the same value as the "index." What's going wrong here? I'm new to this package sorry if the solution is obvious!

**Comment by giswqs** [comment, marked as answer]
The whiteboxR package is only a wrapper of the whitebox-tools backend. This issue is probably related to the backend. You can open an issue at https://github.com/jblindsay/whitebox-tools/issues.

**Reply by brownag** [reply]
Yes, the relevant whitebox-tools error is:
```
thread 'main' panicked at 'index out of bounds: the len is 1458 but the index is 1458', whitebox-lidar\src\las.rs:461:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The R warning message is a by product of logic in the way the results get printed to console, but are unrelated to the core problem (the tool fails to run completely with above error)

**Reply by mereiser** [reply]
That makes sense, thanks! Just asked my question over on whitebox-tools.
